File: src/org/genyris/io/InStreamEOF.java
// Copyright 2008 Peter William Birch <birchb@genyis.org>
//
// This software may be used and distributed according to the terms
// of the Genyris License, in the file "LICENSE", incorporated herein by reference.
//
package org.genyris.io;

import org.genyris.exception.GenyrisException;


public interface InStreamEOF {

    public static final int EOF = -1;

    int getChar()  throws LexException;                // returns EOF on end of file.

    void close()  throws GenyrisException ;

	void resetAfterError();

}

File: src/org/genyris/io/InStreamEOF.kt
// Copyright 2008 Peter William Birch <birchb@genyis.org>
//
// This software may be used and distributed according to the terms
// of the Genyris License, in the file "LICENSE", incorporated herein by reference.
//
package org.genyris.io

import org.genyris.exception.GenyrisException
import org.genyris.interp.Environment

interface InStreamEOF {
    @get:kotlin.Throws(LexException::class)
    val char: Int

    @kotlin.Throws(GenyrisException::class)
    fun close()

    @kotlin.Throws(LexException::class)
    fun resetAfterError()

    fun withinExpression(env: Environment?)

    fun beginningExpression()

    val lineNUmber: Int

    val filename: String?

    companion object {
        val EOF: Int = -1
    }
}
